pallets/loans/src/tests/transfer_debt.rs

```diff
@@ -0,0 +1,425 @@
+use super::*;
+
+fn config_mocks(
+	transfer: (
+		LoanId,
+		LoanId,
+		RepaidPricingAmount<Runtime>,
+		PricingAmount<Runtime>,
+	),
+) {
+	MockPrices::mock_get(|id, pool_id| {
+		assert_eq!(*id, REGISTER_PRICE_ID);
+		assert_eq!(*pool_id, POOL_A);
+		Ok((PRICE_VALUE, BLOCK_TIME.as_secs()))
+	});
+	MockPrices::mock_register_id(|id, pool_id| {
+		assert_eq!(*pool_id, POOL_A);
+		assert_eq!(*id, REGISTER_PRICE_ID);
+		Ok(())
+	});
+	MockChangeGuard::mock_note({
+		let (loan_1, loan_2, repay, borrow) = transfer.clone();
+		move |pool_id, change| {
+			assert_eq!(pool_id, POOL_A);
+			assert_eq!(
+				change,
+				Change::TransferDebt(loan_1, loan_2, repay.clone(), borrow.clone())
+			);
+			Ok(CHANGE_ID)
+		}
+	});
+	MockChangeGuard::mock_released({
+		let (loan_1, loan_2, repay, borrow) = transfer.clone();
+		move |pool_id, change_id| {
+			assert_eq!(pool_id, POOL_A);
+			assert_eq!(change_id, CHANGE_ID);
+			Ok(Change::TransferDebt(
+				loan_1,
+				loan_2,
+				repay.clone(),
+				borrow.clone(),
+			))
+		}
+	});
+}
+
+#[test]
+fn with_wrong_borrower() {
```

Comment:
Maybe would be good to check this in both directions, that the origin needs to be a borrower of both the `from` and `to` loans

Reply:
If I understand you well, I think it works just like you say now, because both loans are from another borrower different than the origin used in `propose_transfer_debt()`. But I think it could be interesting to check what happens if only one of the loans is different from the origin used. 

Added a couple of extra tests regarding this situation.